designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputAge.kt

```diff
@@ -118,10 +124,15 @@ fun InputAge(
         }
     }
 
+    val supportingText = provideSupportingText(uiModel, selectableDates)
+
     InputShell(
         modifier = modifier.testTag("INPUT_AGE").focusRequester(focusRequester),
         title = uiModel.title,
-        state = uiModel.state,
+        state = when (supportingText) {
+            uiModel.supportingText -> uiModel.state
+            else -> InputShellState.ERROR
+        },
```

Comment:
would this entail that if there is no supportng text the input would always show an error if there is no state? should it be Unfocused by default?

Reply:
No, it will only display the error when the supporting text will be different of the model, that is the case the supporting text display the future dates message